Refactors build_features to improve performance (#2224)
alteryx/featuretools: docs/source/release_notes.rst

Future Release
==============
    * Enhancements
        * Refactor build_features code path for performance (:pr:`2224`)
    * Fixes
    * Changes

>         * Add tests/profiling/dfs_profile.py (:pr:`2224`)

Reviewer: This should probably be in the testing changes section.

Author: will do

Future Release
==============
    * Enhancements
        * Refactor build_features code path for performance (:pr:`2224`)
    * Fixes
    * Changes
        * Add Dockerfile.arm for Mac M1 testing/debugging (:pr:`2224`)
        * Add tests/profiling/dfs_profile.py (:pr:`2224`)
    * Documentation Changes
    * Testing Changes
        * Update create feedstock PR forked repo to use (:pr:`2223`)

    Thanks to the following people for contributing to this release:

>     :user:`gsheni`

Reviewer 2: ```suggestion
    :user:`cp2boston`, :user:`gsheni`
```

Author: +1

>     :user:`cpark`, :user:`gsheni`

Reviewer: Correct user name?
<img width="353" alt="image" src="https://user-images.githubusercontent.com/4307001/186427238-a0aa723a-9b3c-49ab-867a-4153e88dbfbe.png">

>     :user:`cpark`, :user:`gsheni`

Reviewer 3: ```suggestion
    :user:`cp2boston`, :user:`gsheni`
```

Future Release
==============
    * Enhancements

>         * Refactor build_features code path for performance (:pr:`2224`)

Reviewer 3: ```suggestion
        * Refactor build_features to speed up long running DFS calls by 50% (:pr:`2224`)
```

Future Release
==============
    * Enhancements
        * Refactor build_features code path for performance (:pr:`2224`)
    * Fixes

>         * Allow boolean columns to be included in remove_highly_correlated_features (:pr:`2231`)

Author: I think merge from main got confused.  Added it back in

        * Refactor schema version checking to use `packaging` method (:pr:`2230`)
        * Extract duplicated logic for Rolling primitives into a general utility function (:pr:`2218`)
    * Documentation Changes
    * Testing Changes
        * Add tests/profiling/dfs_profile.py (:pr:`2224`)
        * Update create feedstock PR forked repo to use (:pr:`2223`, :pr:`2237`)
        * Update development requirements and use latest for documentation (:pr:`2225`)

    Thanks to the following people for contributing to this release:

>     :user:`cp2boston`, :user:`gsheni`, :user:`ozzieD`, ::user:`sbadithe`, :user:`tamargrey`

Author: fixed

>     :user:`cp2boston`, :user:`gsheni`, :user:`ozzieD`, ::user:`sbadithe`, :user:`tamargrey`

Author: Fixed
